### ISSUE TITLE
BUG: kernel NULL pointer dereference: tfw_hpack_set_entry

### ISSUE THREAD

**Issue by EvgeniiMekhanik** [tempesta-tech/tempesta#2187]
```
tempesta login: [17411.944170] BUG: kernel NULL pointer dereference, address: 0000000000000001
[17411.953441] #PF: supervisor read access in kernel mode
[17411.955857] #PF: error_code(0x0000) - not-present page
[17411.958242] PGD 0 P4D 0 
[17411.960303] Oops: 0000 [#1] SMP PTI
[17411.962576] CPU: 2 PID: 0 Comm: swapper/2 Tainted: G        W  OE     5.10.35+ #4
[17411.965271] Hardware name: QEMU Standard PC (Q35 + ICH9, 2009), BIOS 1.15.0-1 04/01/2014
[17411.968027] RIP: 0010:__memcpy_fast+0x13d/0x160 [tempesta_lib]
[17411.970340] Code: 83 c0 08 48 83 c1 08 48 89 70 f8 f6 c2 04 0f 84 25 ff ff ff 8b 31 48 83 c0 04 48 83 c1 04 89 70 fc f6 c2 02 0f 84 18 ff ff ff <0f> b7 31 48 83 c0 02 48 83 c1 02 66 89 70 fe 83 e2 01 74 05 0f b6
[17411.976014] RSP: 0018:ffffba9e401b4738 EFLAGS: 00010202
[17411.978254] RAX: ffff9bab695e205d RBX: ffff9babf4897600 RCX: 0000000000000001
[17411.980907] RDX: 0000000000000003 RSI: 0000000000000001 RDI: ffff9bab695e205d
[17411.983379] RBP: ffffba9e401b4740 R08: 0000000000000001 R09: ffff9bab68c7e900
[17411.985934] R10: ffff9babcf4ab020 R11: ffff9babf4897010 R12: ffff9babcf4ab040
[17411.988454] R13: ffff9babcf4ab040 R14: ffff9bab695e205d R15: ffff9babf4897020
[17411.990806] FS:  0000000000000000(0000) GS:ffff9baeaf680000(0000) knlGS:0000000000000000
[17411.993389] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
[17411.995511] CR2: 0000000000000001 CR3: 00000001239e2006 CR4: 0000000000770ee0
[17411.997873] DR0: 0000000000000000 DR1: 0000000000000000 DR2: 0000000000000000
[17412.000202] DR3: 0000000000000000 DR6: 00000000fffe0ff0 DR7: 0000000000000400
[17412.002455] PKRU: 55555554
[17412.004133] Call Trace:
[17412.005790]  <IRQ>
[17412.007309]  ? memcpy_fast+0xe/0x10 [tempesta_lib]
[17412.009251]  tfw_hpack_decode+0x1532/0x2300 [tempesta_fw]
[17412.011150]  tfw_h2_parse_req+0x1a6/0x320 [tempesta_fw]
[17412.013124]  ss_skb_process+0xf5/0x140 [tempesta_fw]
[17412.014877]  ? h2_set_hdr_x_method_override+0x20/0x20 [tempesta_fw]
[17412.016951]  tfw_http_req_process+0x97/0xa40 [tempesta_fw]
[17412.018716]  ? __tfw_pool_new+0x24/0x70 [tempesta_fw]
[17412.020590]  ? bzero_fast+0xe/0x10 [tempesta_lib]
[17412.022263]  ? __tfw_http_msg_alloc+0x25b/0x340 [tempesta_fw]
[17412.024146]  ? bzero_fast+0xe/0x10 [tempesta_lib]
[17412.025815]  ? tfw_http_init_parser_req+0x25/0x60 [tempesta_fw]
[17412.027617]  ? __tfw_pool_new+0x24/0x70 [tempesta_fw]
[17412.029352]  tfw_http_msg_process_generic+0x188/0x6e0 [tempesta_fw]
[17412.031138]  ? ss_skb_process+0xf5/0x140 [tempesta_fw]
[17412.032863]  ? tfw_h2_send_rst_stream+0xb0/0xb0 [tempesta_fw]
[17412.034509]  tfw_h2_frame_process+0x42b/0x6b0 [tempesta_fw]
[17412.036235]  tfw_http_msg_process+0x48/0x60 [tempesta_fw]
[17412.037848]  tfw_connection_recv+0xbb/0x140 [tempesta_fw]
[17412.039424]  tfw_tls_connection_recv+0x332/0x450 [tempesta_fw]
[17412.041129]  ss_tcp_process_data+0x20a/0x4b0 [tempesta_fw]
[17412.042625]  ss_tcp_data_ready+0x57/0x140 [tempesta_fw]
[17412.044209]  tcp_data_ready+0x2b/0xd0
[17412.045484]  tcp_data_queue+0x805/0xe50
[17412.046696]  tcp_rcv_established+0x254/0x910
[17412.048078]  tcp_v4_do_rcv+0x140/0x200
[17412.049325]  tcp_v4_rcv+0xcd0/0xe20
[17412.050452]  ip_protocol_deliver_rcu+0x44/0x230
[17412.051815]  ip_local_deliver_finish+0x48/0x60
[17412.053173]  ip_local_deliver+0x70/0x110
[17412.054346]  ? ip_rcv_finish_core.constprop.0+0x172/0x470
[17412.055813]  ip_rcv_finish+0x87/0xa0
[17412.057027]  ip_rcv+0xce/0xe0
[17412.058033]  ? ip_rcv_finish_core.constprop.0+0x470/0x470
[17412.059408]  __netif_receive_skb_one_core+0x86/0xa0
[17412.060764]  __netif_receive_skb+0x18/0x60
[17412.061920]  process_backlog+0x9e/0x170
[17412.063033]  net_rx_action+0x13b/0x430
[17412.064198]  __do_softirq+0xe3/0x340
[17412.065312]  asm_call_irq_on_stack+0x12/0x20
[17412.066512]  </IRQ>
[17412.067400]  do_softirq_own_stack+0x3d/0x50
[17412.068613]  irq_exit_rcu+0xa2/0xe0
[17412.069713]  sysvec_call_function_single+0x3d/0x90
[17412.070942]  asm_sysvec_call_function_single+0x12/0x20
[17412.072285] RIP: 0010:native_safe_halt+0xe/0x10
[17412.073502] Code: 39 ff ff ff 4c 89 ee 48 c7 c7 a0 ba 85 b0 e8 89 64 91 ff e9 01 ff ff ff cc cc cc cc e9 07 00 00 00 0f 00 2d 96 55 47 00 fb f4 <c3> 90 e9 07 00 00 00 0f 00 2d 86 55 47 00 f4 c3 cc cc 0f 1f 44 00
[17412.077365] RSP: 0018:ffffba9e40093e88 EFLAGS: 00000212
[17412.078659] RAX: ffffffffaf996750 RBX: 0000000000000002 RCX: ffff9baeaf6acdc0
[17412.080377] RDX: 000000000081d2ae RSI: 0000000000000083 RDI: 0000000000000002
[17412.081996] RBP: ffffba9e40093e90 R08: ffffffffb0293970 R09: 0000000000000000
[17412.083672] R10: 00000fd605dd8019 R11: 0000000000000000 R12: ffff9bab6034dc40
[17412.085349] R13: 0000000000000000 R14: 0000000000000000 R15: 0000000000000000
[17412.086935]  ? __sched_text_end+0x4/0x4
[17412.088149]  ? default_idle+0xe/0x20
[17412.089320]  arch_cpu_idle+0x15/0x20
[17412.090436]  default_idle_call+0x3d/0xc0
[17412.091629]  do_idle+0x215/0x2a0
[17412.092749]  cpu_startup_entry+0x20/0x30
[17412.093883]  start_secondary+0x145/0x1b0
[17412.095017]  secondary_startup_64_no_verify+0xc2/0xcb
```

e42225f5

```asm
/home/kingluo/tempesta/lib/str_simd.S:127
.L2cpy:
        movzwl  (%rcx), %esi
 9ed:   0f b7 31                movzwl (%rcx),%esi ;<--------- %rcx=input_%rsi=s->data=0x01
...
/home/kingluo/tempesta/fw/hpack.c:741
                memcpy_fast(data, s->data, s->len);
   1b5e9:       e8 00 00 00 00          call   1b5ee <tfw_hpack_decode+0x156e>
                        1b5ea: R_X86_64_PLT32   __asan_load8_noabort-0x4
   1b5ee:       49 8b 56 e8             mov    -0x18(%r14),%rdx
   1b5f2:       49 8b 76 e0             mov    -0x20(%r14),%rsi  ; <----------- s->data
   1b5f6:       4c 89 e7                mov    %r12,%rdi
   1b5f9:       e8 00 00 00 00          call   1b5fe <tfw_hpack_decode+0x157e>
                        1b5fa: R_X86_64_PLT32   memcpy_fast-0x4

```

**Comment by EvgeniiMekhanik** [comment]
```
listen 192.168.122.100:443 proto=h2,https;
listen 192.168.122.100:80 proto=http;

access_log on;
client_tbl_size 134217728;

block_action attack reply;
block_action error reply;

tls_certificate /home/tempesta/certs/tempesta.kinescope.io.pem;
tls_certificate_key /home/tempesta/certs/tempesta.kinescope.io.key;

srv_group main {
	server 192.168.122.1:443;
}

srv_group http {
	server 192.168.122.1:80;
}

vhost main {
        proxy_pass main;
}

vhost http {
	proxy_pass http;
}

http_chain {
	mark == 1 -> http;
	-> main;
}
```

**Comment by EvgeniiMekhanik** [comment]
Another dump
```
[12622.176271] BUG: kernel NULL pointer dereference, address: 0000000000000008
[12622.183542] #PF: supervisor read access in kernel mode
[12622.185797] #PF: error_code(0x0000) - not-present page
[12622.188109] PGD 0 P4D 0
[12622.190019] Oops: 0000 [#1] SMP PTI
[12622.191983] CPU: 15 PID: 0 Comm: swapper/15 Tainted: G           OE     5.10.35+ #4
[12622.194615] Hardware name: QEMU Standard PC (Q35 + ICH9, 2009), BIOS 1.15.0-1 04/01/2014
[12622.197185] RIP: 0010:tfw_hpack_decode+0xcbf/0x2300 [tempesta_fw]
[12622.199473] Code: 4c 03 65 90 eb 1d 83 c3 01 49 83 c4 70 41 83 ee 01 39 5d a0 0f 84 6c 0a 00 00 45 39 ef 0f 83 0d 0a 00 00 49 8b 34 24 44 89 c7 <2b> 7e 08 41 01 fd 41 f6 44 24 1c 01 74 ce 49 8b 79 50 4c 89 4d 88
[12622.204951] RSP: 0018:ffffa4cc40458718 EFLAGS: 00010282
[12622.207185] RAX: 0000000000000000 RBX: 0000000000000000 RCX: 000000000000009d
[12622.209526] RDX: 0000000000000f63 RSI: 0000000000000000 RDI: 00000000ffffffe0
[12622.212066] RBP: ffffa4cc404587b8 R08: 00000000ffffffe0 R09: ffff99046ed208f8
[12622.214523] R10: 0000000000000000 R11: ffff9903eb78610b R12: ffff9903ebd62010
[12622.216944] R13: 0000000000000fd7 R14: 0000000000000021 R15: 0000000000000f63
[12622.219317] FS:  0000000000000000(0000) GS:ffff99072f9c0000(0000) knlGS:0000000000000000
[12622.221814] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
[12622.223963] CR2: 0000000000000008 CR3: 00000001b1112002 CR4: 0000000000770ee0
[12622.226304] DR0: 0000000000000000 DR1: 0000000000000000 DR2: 0000000000000000
[12622.228598] DR3: 0000000000000000 DR6: 00000000fffe0ff0 DR7: 0000000000000400
[12622.230889] PKRU: 55555554
[12622.232572] Call Trace:
[12622.234222]  <IRQ>
[12622.235749]  tfw_h2_parse_req+0x1a6/0x320 [tempesta_fw]
[12622.237680]  ss_skb_process+0xf5/0x140 [tempesta_fw]
[12622.239553]  ? h2_set_hdr_x_method_override+0x20/0x20 [tempesta_fw]
[12622.241582]  ? h2_set_hdr_x_method_override+0x20/0x20 [tempesta_fw]
[12622.243604]  tfw_http_req_process+0x9b/0xa60 [tempesta_fw]
[12622.245370]  ? tfw_pool_alloc_pages+0x53/0x60 [tempesta_fw]
[12622.247286]  ? __get_free_pages+0x11/0x40
[12622.248974]  ? tfw_pool_alloc_pages+0x53/0x60 [tempesta_fw]
[12622.249300] [tempesta fw] 172.115.248.175 "player-metrics.kinescope.io.vhost_main" "POST /v1/batch/gzip?referrer=https%3A%2F%2Frevitonica.ru%2F HTTP/2.0" 200 0 "https://kinescope.io/" "Mozilla/5.0 (X11; CrOS x86_64 14541.0.0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/126.0.0.0 Safari/537.36"
[12622.250871]  ? __tfw_pool_new+0x24/0x70 [tempesta_fw]
[12622.259429]  tfw_http_msg_process_generic+0x1c3/0x900 [tempesta_fw]
[12622.261407]  ? __put_page+0x4f/0x90
[12622.262967]  ? memcpy_fast+0xe/0x10 [tempesta_lib]
[12622.264648]  tfw_h2_frame_process+0x42b/0x6b0 [tempesta_fw]
[12622.266039] [tempesta fw] 188.170.83.37 "player-metrics.kinescope.io.vhost_main" 
```

**Comment by kingluo** [comment]
```
[ 9241.758089] BUG: kernel NULL pointer dereference, address: 0000000000000008
[ 9241.759403] #PF: supervisor read access in kernel mode
[ 9241.760288] #PF: error_code(0x0000) - not-present page
[ 9241.761169] PGD 0 P4D 0 
[ 9241.761700] Oops: 0000 [#1] SMP KASAN PTI
[ 9241.762487] CPU: 6 PID: 0 Comm: swapper/6 Tainted: G    B      OE     5.10.35+ #5
[ 9241.763861] Hardware name: QEMU Standard PC (Q35 + ICH9, 2009), BIOS 1.15.0-1 04/01/2014
[ 9241.765441] RIP: 0010:tfw_hpack_decode+0x1940/0x3c50 [tempesta_fw]
[ 9241.784017] Code: 10 00 00 44 39 65 b0 0f 83 10 10 00 00 48 89 df e8 b5 28 9e d9 4c 8b 2b 49 8d 7d 08 e8 a9 28 9e d9 ba e0 ff ff ff 48 8d 7b 1c <41> 2b 55 08 41 01 d4 e8 34 25 9e d9 f6 43 1c 01 74 b1 48 8b bd 68
[ 9241.787148] RSP: 0018:ffff888420188e50 EFLAGS: 00010282
[ 9241.788061] RAX: 0000000000000000 RBX: ffff8881cd93c010 RCX: ffffffffc1717377
[ 9241.789274] RDX: 00000000ffffffe0 RSI: 0000000000000008 RDI: ffff8881cd93c02c
[ 9241.790589] RBP: ffff888420188f48 R08: ffffffffc1716d49 R09: ffff8881e3d860c7
[ 9241.797484] R10: 00000000000000ac R11: 0000000000000001 R12: 0000000000000fcd
[ 9241.797486] R13: 0000000000000000 R14: 0000000000000041 R15: 0000000000000000
[ 9241.797491] FS:  0000000000000000(0000) GS:ffff888420180000(0000) knlGS:0000000000000000
[ 9241.834645] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
[ 9241.834648] CR2: 0000000000000008 CR3: 00000001d941e005 CR4: 0000000000770ee0
[ 9241.834652] DR0: 0000000000000000 DR1: 0000000000000000 DR2: 0000000000000000
[ 9241.834657] DR3: 0000000000000000 DR6: 00000000fffe0ff0 DR7: 0000000000000400
[ 9241.868321] PKRU: 55555554
[ 9241.908572] Call Trace:
[ 9241.908575]  <IRQ>
[ 9241.908631]  ? create_prof_cpu_mask+0x30/0x30
[ 9241.908679]  tfw_h2_parse_req+0x382/0x590 [tempesta_fw]
[ 9241.953404]  ? tcp_data_ready+0x73/0x210
[ 9242.000530]  ? tcp_rcv_established+0x98f/0xf70
[ 9242.000537]  ? tcp_v4_do_rcv+0x25f/0x360
[ 9242.034845]  ss_skb_process+0x26f/0x2f0 [tempesta_fw]
[ 9242.034889]  ? h2_set_hdr_x_method_override+0x50/0x50 [tempesta_fw]
[ 9242.088834]  ? tfw_sock_srv_exit+0x30/0x30 [tempesta_fw]
[ 9242.123983]  ? prep_new_page+0xae/0x110
[ 9242.171167]  tfw_http_req_process+0x145/0x1120 [tempesta_fw]
[ 9242.223490]  ? tfw_http_extract_request_authority+0x110/0x110 [tempesta_fw]
[ 9242.285011]  ? rb_insert_color+0x10b/0x2a0
[ 9242.330851]  ? memset+0x3c/0x50
[ 9242.330871]  ? bzero_fast+0xe/0x10 [tempesta_lib]
[ 9242.386895]  ? tfw_pool_alloc_pages+0x51/0x80 [tempesta_fw]
[ 9242.386902]  ? __kasan_check_write+0x14/0x20
[ 9242.439885]  ? tfw_http_conn_msg_alloc+0x378/0x3c0 [tempesta_fw]
[ 9242.439925]  tfw_http_msg_process_generic+0x36b/0xec0 [tempesta_fw]
[ 9242.494073]  ? mem_cgroup_calculate_protection+0x1e0/0x1e0
[ 9242.494081]  ? free_unref_page_commit+0x12b/0x1b0
[ 9242.548291]  ? tfw_http_req_process+0x1120/0x1120 [tempesta_fw]
[ 9242.585885]  ? __kasan_check_write+0x14/0x20
[ 9242.585932]  ? ss_skb_chop_head_tail+0x31b/0x3a0 [tempesta_fw]
[ 9242.585938]  ? __kasan_check_write+0x14/0x20
[ 9242.622419]  ? ss_skb_expand_head_tail+0x180/0x180 [tempesta_fw]
[ 9242.622426]  ? memcpy_fast+0xe/0x10 [tempesta_lib]
[ 9242.678725]  ? ss_skb_split+0x1de/0x220 [tempesta_fw]
[ 9242.733512]  tfw_h2_frame_process+0x7c1/0xc80 [tempesta_fw]
[ 9242.733553]  ? tfw_h2_frame_recv+0x3940/0x3940 [tempesta_fw]
[ 9242.770382]  ? ss_skb_chop_head_tail+0x2a0/0x3a0 [tempesta_fw]
[ 9242.770424]  tfw_http_msg_process+0x9e/0xc0 [tempesta_fw]
[ 9242.826846]  tfw_connection_recv+0x188/0x250 [tempesta_fw]
[ 9242.826886]  ? tfw_connection_send+0x60/0x60 [tempesta_fw]
[ 9242.885701]  ? ss_skb_list_chop_head_tail+0x6e/0x1d0 [tempesta_fw]
[ 9242.885754]  ? ttls_payload_off+0x3f/0x1a0 [tempesta_tls]
[ 9242.943426]  tfw_tls_connection_recv+0x52f/0x700 [tempesta_fw]
[ 9242.943468]  ? tfw_tls_connection_lost+0x40/0x40 [tempesta_fw]
[ 9242.981621]  ss_tcp_process_data+0x45f/0x8a0 [tempesta_fw]
[ 9242.981670]  ? ss_linkerror+0x90/0x90 [tempesta_fw]
[ 9243.019458]  ? __alloc_skb+0x272/0x330
[ 9243.019500]  ss_tcp_data_ready+0xa9/0x1d0 [tempesta_fw]
[ 9243.077577]  tcp_data_ready+0x73/0x210
[ 9243.077584]  tcp_rcv_established+0x98f/0xf70
[ 9243.144385]  ? tcp_data_queue+0x1e00/0x1e00
[ 9243.144391]  ? __kasan_check_read+0x11/0x20
[ 9243.183788]  tcp_v4_do_rcv+0x25f/0x360
[ 9243.222394]  tcp_v4_rcv+0x1579/0x16b0
[ 9243.222401]  ? tcp_v4_early_demux+0x300/0x300
[ 9243.306268]  ? __kasan_check_write+0x14/0x20
[ 9243.363206]  ip_protocol_deliver_rcu+0x57/0x340
[ 9243.363213]  ip_local_deliver_finish+0xc6/0xf0
[ 9243.419526]  ip_local_deliver+0x136/0x210
[ 9243.419532]  ? ip_local_deliver_finish+0xf0/0xf0
[ 9243.436545]  ? tcp_v4_early_demux+0x2a0/0x300
[ 9243.492905]  ? ip_rcv_finish_core.constprop.0+0x17c/0x8d0
[ 9243.492912]  ip_rcv_finish+0xcf/0xf0
[ 9243.547894]  ip_rcv+0x16d/0x180
[ 9243.630900]  ? ip_local_deliver+0x210/0x210
[ 9243.648692]  ? ip_rcv_finish_core.constprop.0+0x8d0/0x8d0
[ 9243.686050]  ? ip_local_deliver+0x210/0x210
[ 9243.686062]  __netif_receive_skb_one_core+0x132/0x140
[ 9243.686068]  ? __netif_receive_skb_core+0x1900/0x1900
[ 9243.722928]  ? find_next_bit+0x14/0x20
[ 9243.786759]  ? cpumask_next+0x2c/0x40
[ 9243.786765]  ? __kasan_check_write+0x14/0x20
[ 9243.822018]  ? _raw_spin_lock+0x7b/0xd0
[ 9243.822026]  __netif_receive_skb+0x26/0xb0
[ 9243.858031]  process_backlog+0xfe/0x290
[ 9243.858038]  net_rx_action+0x287/0x730
[ 9243.912956]  ? napi_complete_done+0x2c0/0x2c0
[ 9243.912986]  ? switch_fpu_return+0x120/0x120
[ 9243.912994]  __do_softirq+0x106/0x445
[ 9243.958392]  asm_call_irq_on_stack+0x12/0x20
[ 9243.983083]  </IRQ>
[ 9243.983095]  do_softirq_own_stack+0x3d/0x50
[ 9243.983108]  irq_exit_rcu+0xcf/0x120
[ 9244.074665]  sysvec_call_function_single+0x3a/0x90
[ 9244.127995]  asm_sysvec_call_function_single+0x12/0x20
[ 9244.128010] RIP: 0010:native_safe_halt+0xe/0x10
[ 9244.128016] Code: 56 ff e9 a8 fe ff ff 4c 89 e6 48 c7 c7 60 6b 51 9d e8 b6 6e 56 ff e9 70 fe ff ff cc e9 07 00 00 00 0f 00 2d 04 fc 55 00 fb f4 <c3> 90 e9 07 00 00 00 0f 00 2d f4 fb 55 00 f4 c3 cc cc 0f 1f 44 00
[ 9244.162871] RSP: 0018:ffff88810e42fdb8 EFLAGS: 00000206
[ 9244.205059] 
[ 9244.205065] RAX: ffffffff9c0fb260 RBX: ffff88810e423f80 RCX: ffffffff9c0e40c5
[ 9244.221866] RDX: 000000000027cb72 RSI: 0000000000000004 RDI: ffff8884201b5040
[ 9244.276394] RBP: ffff88810e42fdc0 R08: 0000000000000001 R09: ffff8884201b5043
[ 9244.276396] R10: ffffed1084036a08 R11: 0000000000000001 R12: 0000000000000006
[ 9244.276399] R13: ffffffff9e2a1be0 R14: 0000000000000000 R15: 0000000000000000
[ 9244.276407]  ? __cpuidle_text_start+0x8/0x8
[ 9244.331637]  ? rcu_eqs_enter.constprop.0+0x85/0xa0
[ 9244.331643]  ? default_idle+0xe/0x20
[ 9244.331648]  arch_cpu_idle+0x15/0x20
[ 9244.331655]  default_idle_call+0x66/0x160
[ 9244.385748]  do_idle+0x379/0x440
[ 9244.385756]  ? arch_cpu_idle_exit+0x40/0x40
[ 9244.440483]  cpu_startup_entry+0x20/0x30
[ 9244.497019]  start_secondary+0x1f2/0x270
[ 9244.497025]  ? set_cpu_sibling_map+0x18e0/0x18e0
[ 9244.553786]  ? set_bringup_idt_handler.constprop.0+0x84/0x90
[ 9244.553793]  ? start_cpu0+0xc/0xc
[ 9244.610745]  secondary_startup_64_no_verify+0xc2/0xcb
[ 9244.610759] Modules linked in:
[ 9244.674949]  nft_counter
[ 9244.762590]  xt_mark xt_tcpudp nft_compat nf_tables nfnetlink
[ 9244.820608]  tempesta_fw(OE)
[ 9244.881010]  tempesta_db(OE)
[ 9244.920396]  sha256_ssse3 sha512_ssse3 tempesta_tls(OE) tempesta_lib(OE) intel_rapl_msr
[ 9244.957618]  intel_rapl_common
[ 9245.017055]  isst_if_common
[ 9245.054945]  nfit kvm_intel snd_hda_codec_generic ledtrig_audio
[ 9245.092112]  snd_hda_intel snd_intel_dspcfg kvm
[ 9245.149461]  binfmt_misc snd_hda_codec
[ 9245.216173]  snd_hda_core rapl
[ 9245.234582]  snd_hwdep
[ 9245.291962]  snd_pcm
[ 9245.328445]  snd_timer
[ 9245.365015]  joydev input_leds
[ 9245.429551]  snd serio_raw
[ 9245.486354]  soundcore
[ 9245.523871]  qemu_fw_cfg
[ 9245.560398]  mac_hid
[ 9245.622464]  dm_multipath
[ 9245.657849]  sch_fq_codel scsi_dh_rdac
[ 9245.715406]  scsi_dh_emc
[ 9245.772040]  scsi_dh_alua
[ 9245.830319]  msr
[ 9245.891313]  efi_pstore ip_tables x_tables
[ 9245.952654]  autofs4
[ 9246.011960]  btrfs
[ 9246.071783]  blake2b_generic raid10 raid456
[ 9246.135236]  async_raid6_recov
[ 9246.196239]  async_memcpy async_pq
[ 9246.257381]  async_xor async_tx
[ 9246.319040]  xor
[ 9246.388089]  raid6_pq
[ 9246.449572]  libcrc32c raid1
[ 9246.509377]  raid0
[ 9246.567805]  multipath
[ 9246.629433]  linear hid_generic
[ 9246.647844]  qxl
[ 9246.706470]  usbhid
[ 9246.734458]  hid
[ 9246.792285]  drm_ttm_helper
[ 9246.849011]  ttm drm_kms_helper syscopyarea
[ 9246.886742]  sysfillrect
[ 9246.923378]  sysimgblt
[ 9246.980204]  fb_sys_fops
[ 9246.997365]  cec
[ 9247.032917]  rc_core
[ 9247.088048]  drm crct10dif_pclmul crc32_pclmul ghash_clmulni_intel
[ 9247.144907]  aesni_intel psmouse crypto_simd cryptd virtio_net ahci glue_helper
[ 9247.215294]  i2c_i801 libahci
[ 9247.297078]  net_failover
[ 9247.331448]  i2c_smbus
[ 9247.386986]  lpc_ich
[ 9247.424323]  xhci_pci
[ 9247.480766]  virtio_blk virtio_rng
[ 9247.518765]  failover
[ 9247.575843]  xhci_pci_renesas
[ 9247.592692] 
[ 9247.592718] CR2: 0000000000000008
[ 9247.629546] ---[ end trace 240ebb03671f1790 ]---
[ 9247.665603] RIP: 0010:tfw_hpack_decode+0x1940/0x3c50 [tempesta_fw]
[ 9247.665611] Code: 10 00 00 44 39 65 b0 0f 83 10 10 00 00 48 89 df e8 b5 28 9e d9 4c 8b 2b 49 8d 7d 08 e8 a9 28 9e d9 ba e0 ff ff ff 48 8d 7b 1c <41> 2b 55 08 41 01 d4 e8 34 25 9e d9 f6 43 1c 01 74 b1 48 8b bd 68
[ 9247.721243] RSP: 0018:ffff888420188e50 EFLAGS: 00010282
[ 9247.721249] RAX: 0000000000000000 RBX: ffff8881cd93c010 RCX: ffffffffc1717377
[ 9247.721256] RDX: 00000000ffffffe0 RSI: 0000000000000008 RDI: ffff8881cd93c02c
[ 9247.776833] RBP: ffff888420188f48 R08: ffffffffc1716d49 R09: ffff8881e3d860c7
[ 9247.776835] R10: 00000000000000ac R11: 0000000000000001 R12: 0000000000000fcd
[ 9247.776837] R13: 0000000000000000 R14: 0000000000000041 R15: 0000000000000000
[ 9247.776844] FS:  0000000000000000(0000) GS:ffff888420180000(0000) knlGS:0000000000000000
[ 9247.821176] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
[ 9247.821185] CR2: 0000000000000008 CR3: 00000001d941e005 CR4: 0000000000770ee0
[ 9247.821189] DR0: 0000000000000000 DR1: 0000000000000000 DR2: 0000000000000000
[ 9247.821191] DR3: 0000000000000000 DR6: 00000000fffe0ff0 DR7: 0000000000000400
[ 9247.821193] PKRU: 55555554
[ 9247.821196] Kernel panic - not syncing: Fatal exception in interrupt
[ 9247.848380] Kernel Offset: 0x19c00000 from 0xffffffff81000000 (relocation range: 0xffffffff80000000-0xffffffffbfffffff)
[ 9258.811503] Rebooting in 1 seconds..
```

**Comment by krizhanovsky** [comment]
Fixed by #2208

**Comment by EvgeniiMekhanik** [comment]
The problem was in memory corruption in hpack, which corrupt memory in random place.
